core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -321,9 +321,13 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   override def getLastUpdatedTime(): Long = lastScanTime.get()
 
   override def getAppUI(appId: String, attemptId: Option[String]): Option[LoadedAppUI] = {
+    val logPath = RollingEventLogFilesWriter.EVENT_LOG_DIR_NAME_PREFIX +
+        Utils.nameForAppAndAttempt(appId, attemptId)
     val app = try {
       load(appId)
      } catch {
+      case _: NoSuchElementException if this.conf.get(EVENT_LOG_ROLLING_ON_DEMAND_LOAD_ENABLED) =>
```

Comment:
I understand that we are trying to push for usage of `RollingEventLogFilesWriter` as the new default but for users who have single event logs and if they try to get the UI for an app, will this functionality not break for them since `EVENT_LOG_ROLLING_ON_DEMAND_LOAD_ENABLED` is true by default ?

Reply:
Could you elaborate on what we can break here, @thejdeep ?
> will this functionality not break for them since EVENT_LOG_ROLLING_ON_DEMAND_LOAD_ENABLED is true by default ?

Reply:
The dummy metadata is added and cleaned up at `FileNotFound` exception immediate in this function as  @mridulm requested. It works for both non-existing appId and SingleFile logs.